crates/orchestrator/src/jobs/state_update_job/kzg.rs

```diff
@@ -0,0 +1,124 @@
+use crate::data_storage::aws_s3::config::AWSS3Config;
+use crate::data_storage::aws_s3::AWSS3;
+use crate::data_storage::{DataStorage, DataStorageConfig};
+use crate::jobs::state_update_job::CURRENT_PATH;
+use alloy::eips::eip4844::BYTES_PER_BLOB;
+use c_kzg::{Blob, Bytes32, KzgCommitment, KzgProof, KzgSettings};
+
+/// Build KZG proof for a given block
+pub async fn build_kzg_proof(
+    block_number: u64,
+    fetch_from_tests: Option<bool>,
```

Comment:
same as above, `fetch_from_tests` shouldn't be an argument. normal code should almost never change for test cases. during the test case, we should try to mock functions to get the functionality we're looking for. In this specific case, I think mocking `s3_client.get_data` will do the job?